src/state/reducers/students-state.ts

```diff
@@ -0,0 +1,59 @@
+import { nanoid } from 'nanoid'
+
+import {
+  STUDENTS_ADD,
+  STUDENTS_EDIT,
+  STUDENTS_REMOVE,
+  RESET_STUDENTS
+} from "../actions/students";
+
+interface StudentsState {
+  students: Array<{
+    name: string,
+    id: number,
+    count: number
+  }>
+}
+
+const initialState: StudentsState = { students: [] };
+
+export default (state = initialState, action: StudentsTypes): StudentsState => {
```

Comment:
`StudentsTypes` does not seem to be defined - it looks like the type is already created in the `students.ts` file, but requires importing and possibly renaming. Let's do both of those things
```suggestion
export default (state = initialState, action: StudentsActionType): StudentsState => {
```
```ts
// `students-state.ts`
import type { StudentsActionType } from '../actions/students'
```
```ts
// `students.ts`
export const STUDENTS_ADD = "add_student";
export const STUDENTS_EDIT = "edit_student";
export const STUDENTS_REMOVE = "remove_student";
export const RESET_STUDENTS = "reset_students"

export type StudentsActionType =
  | { type: typeof STUDENTS_ADD, name: string }
  | { type: typeof STUDENTS_EDIT, name: string, id: string }
  | { type: typeof STUDENTS_REMOVE, id: string }
  | { type: typeof RESET_STUDENTS }

export const addStudent = (name: string): StudentsActionType => ({
  type: STUDENTS_ADD,
  name
});

export const editStudent = (name: string, id: string): StudentsActionType => ({
  type: STUDENTS_EDIT,
  name,
  id
});

export const removeStudent = (id: string): StudentsActionType => ({
  type: STUDENTS_REMOVE,
  id
});

export const resetStudents = (): StudentsActionType => ({
  type: RESET_STUDENTS,
});
```
Check [this post](https://denistakeda.github.io/articles/003_typescript_adt.html) on _algebraic data types_ implementation in TypeScript with usage of [union types](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#union-types)